src/crypto/CHIPCryptoPAL.h

```diff
@@ -1285,5 +1285,34 @@ enum class MatterOid
  **/
 CHIP_ERROR ExtractDNAttributeFromX509Cert(MatterOid matterOid, const ByteSpan & certificate, uint16_t & id);
 
+class SymmetricKeyContext
```

Comment:
This could probably use documentation.  Does it guarantee use of any particular encryption/decryption algorithms? 

Or put another way, why is the implementation not shared with the only virtual part being getting a reference to the key, if the algorithm is the same for all implementations of this interface?  And if the algorithm is _not_ the same, then that should be clearly documented...

Mostly I'm still a little worried about the separate copies of low-level crypto calls between the group code and the unicast code.  They would ideally just be a single codepath, but I don't see how this approach gets us closer to that.

Reply:
I have added documentation with references to the specification, which states which algorithms must be used. This interface is meant to be used to perform the symmetric key operations without exposing the key value, and it should be used both for the unicast, and multicast paths. The CryptoContext will be modified to use this interface, and its implementation should be used in both cases.